### PR TITLE
Pdf 2

### PR DESCRIPTION
**Pull request by tzhenia**

### CODE REVIEW

**Comment by MSennikov** [comment]
Can you use Lombok?

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/Constants.java [hunk 1]
```diff
@@ -8,4 +8,5 @@
     public static final String REGISTRATION_FAIL_REASON = "Registration Failed! Most likely, a user with such email is already registered.";
     public static final String SHORT_PASSWORD = "Password is too short";
     public static final String CHANGE_ROLE_FAIL = "You can't demote or delete last admin";
+    public static final String Generate_Pdf_FAIL = "Generate PDF Failed!";
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/Constants.java]: `GENERATE_PDF_FAIL` or smth like that

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/Client.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+
+@RestController
+@RequestMapping("/apis/pdf")
+public class Client {
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/Client.java]: Please use this annotation `@RequiredArgsConstructor(onConstructor = @__({@Autowired}))` for class, and remove constructor

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfDownloadController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+@RestController
+@RequestMapping("/api")
+public class PdfDownloadController {
+    @Autowired
+    ServletContext context;
+
+    @Value("${path.for.download}")
+    private String downloadPath;
+    @RequestMapping(value = "/pdf/{fileName:.+}", method = RequestMethod.GET, produces = "application/pdf")
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfDownloadController.java]: `@GetMapping`

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfWritable.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package ua.softserve.rv_028.issuecitymonitor.controller.PDF;
+
+import ua.softserve.rv_028.issuecitymonitor.entity.User;
+
+import java.time.LocalDateTime;
+
+public interface PdfWritable {
+    Long getId();
```
Comment (gefasim) [inline src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfWritable.java]: Maybe better use `default ` parameter for method? 
And you could override only those that you want. 
```
public interface Test {
    default Long getId(){
        return null;
    }
    
    default String getTitle(){
        return null;
    }
    
    default String getCat(){
        return null;
    }    
}
```

**Review by gefasim — CHANGES_REQUESTED** [review]

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfWritable.java (hunk 4 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfWritable.java]: Done!

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfDownloadController.java (hunk 3 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/PdfDownloadController.java]: Done!

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/Constants.java (hunk 1 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/Constants.java]: Done!

File: src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/Client.java (hunk 2 above)
Comment (tzhenia) [reply src/main/java/ua/softserve/rv_028/issuecitymonitor/controller/PDF/Client.java]: Done!

**Review by gefasim — APPROVED** [review]

**Pull request merged**